src/containerPool.ts

```diff
@@ -307,6 +312,41 @@ export class ContainerPool {
     }
   }
 
+  private async cleanupStaleContainers(): Promise<void> {
+    try {
+      // List all containers that match our naming pattern
+      const containers = await this.docker.listContainers({ 
+        all: true,
+        filters: { 
+          name: [`${BASE_NAME}-pool-`, `${BASE_NAME}-`] 
+        }
+      })
+
+      if (containers.length === 0) {
+        return
+      }
+
+      console.warn(`Found ${containers.length} stale zemu containers, cleaning up...`)
+
+      // Remove all found containers in parallel
+      const cleanupPromises = containers.map(async (containerInfo) => {
+        try {
+          const container = this.docker.getContainer(containerInfo.Id)
+          await container.remove({ force: true })
+        } catch (error) {
+          // Ignore errors when removing containers (they might already be removed)
+          console.warn(`Failed to remove stale container ${containerInfo.Names?.[0]}:`, error)
+        }
+      })
+
+      await Promise.all(cleanupPromises)
+      console.warn(`Cleaned up ${containers.length} stale containers`)
+    } catch (error) {
+      console.warn('Failed to cleanup stale containers:', error)
+      // Don't throw - we want initialization to continue even if cleanup fails
```

Comment:
Do we? if cleanup failed, it is for a reason, and it may fail the following tests because of it. Why would we fail to cleanup?